registration/include/pcl/registration/impl/gicp.hpp

```diff
@@ -425,16 +427,90 @@ GeneralizedIterativeClosestPoint<PointSource, PointTarget>::computeTransformatio
   base_transformation_ = Eigen::Matrix4f::Identity();
   nr_iterations_ = 0;
   converged_ = false;
-  double dist_threshold = corr_dist_threshold_ * corr_dist_threshold_;
-  pcl::Indices nn_indices(1);
-  std::vector<float> nn_dists(1);
 
   pcl::transformPointCloud(output, output, guess);
+  pcl::transformPointCloud(output, *input_transformed, transformation_);
+
+  // Make blobs if necessary
+  determineRequiredBlobData();
+  PCLPointCloud2::Ptr target_blob(new PCLPointCloud2);
+  if (need_target_blob_)
+    pcl::toPCLPointCloud2(*target_, *target_blob);
+
+  // Pass in the default target for the Correspondence Estimation/Rejection code
+  correspondence_estimation_->setInputTarget(target_);
+  if (correspondence_estimation_->requiresTargetNormals())
+    correspondence_estimation_->setTargetNormals(target_blob);
+  // Correspondence Rejectors need a binary blob
+  for (std::size_t i = 0; i < correspondence_rejectors_.size(); ++i) {
+    registration::CorrespondenceRejector::Ptr& rej = correspondence_rejectors_[i];
+    if (rej->requiresTargetPoints())
+      rej->setTargetPoints(target_blob);
+    if (rej->requiresTargetNormals() && target_has_normals_)
+      rej->setTargetNormals(target_blob);
+  }
+
+  convergence_criteria_->setMaximumIterations(max_iterations_);
+  //convergence_criteria_->setRelativeMSE(euclidean_fitness_epsilon_);
+  convergence_criteria_->setTranslationThreshold(transformation_epsilon_);
+  convergence_criteria_->setRotationThreshold(rotation_epsilon_);
+
+  do {
+    // Get blob data if needed
+    PCLPointCloud2::Ptr input_transformed_blob;
+    if (need_source_blob_) {
+      input_transformed_blob.reset(new PCLPointCloud2);
+      toPCLPointCloud2(*input_transformed, *input_transformed_blob);
+    }
+
```

Comment:
In ICP, here's:
```cpp
    // Save the previously estimated transformation
    previous_transformation_ = transformation_;
```
But there's `previous_transformation_ = transformation_;` below in GICP, so delete this two lines.